processing-service-core/api/swagger/swagger.yaml

```diff
@@ -952,7 +954,237 @@ paths:
+    get:
+      tags:
+        - Tenants Configuration
+      summary: Retrieve tenant specific config.
+      description: Returns the tenant specific config for this key that is stored against this tenant.
+      operationId: getTenantConfig
+      responses:
+        200:
+          schema: 
+            type: string
+            example: "{'pii.xml': []}"
+          description: Returns the tenant specific configuration stored for the supplied key.
+        400:
+          description: Indicates invalid parameter was passed.
+          schema:
+            $ref: "#/definitions/errorMessage"
+        404:
+          description: Indicates key was not found.
+    delete:
+      tags:
+        - Tenants Configuration
+      summary: Delete tenant specific config for the provided key.
+      description: Delete tenant specific config for the provided key.
+      operationId: deleteTenantConfig
+      responses:
+        204:
+          description: Deletes the tenant specific config for the supplied key.
+        400:
+          description: Indicates invalid parameter was passed.
+          schema:
+            $ref: "#/definitions/errorMessage"
+        404:
+          description: Indicates key was not found.
+  #
+  /tenants/{tenantId}/config:
+    x-swagger-router-controller: tenant
+    parameters:
+      - name: tenantId
+        in: path
+        type: string
+        description: The ID of the tenant.
+        required: true
+    put:
+      tags:
+        - Tenants Configuration
+      parameters:
+        - name: tenantConfigs
+          description: The json string representation key-value pair map of custom tenant configuration.
+          in: body
+          required: true
+          schema:
+            $ref: "#/definitions/tenantConfigs"
+      summary: Bulk method used to create or update multiple tenant specific configs.
+      description: Creates or updates a custom configuration setting for this key against the tenant.
+      operationId: setTenantConfigs
+      responses:
+        204:
+          description: Indicates that the tenant specific configuration was set successfully.
+        400:
+          description: Indicates invalid parameter was passed.
+          schema:
+            $ref: "#/definitions/errorMessage"        
+    get:
+      tags:
+        - Tenants Configuration
+      summary: Retrieve tenant specific configs as map of key-value pairs.
+      description: Returns all tenant specific configs as a list.
+      operationId: getTenantConfigs
+      responses:
+        200:
+          description: |
+            Returns the map of tenant specific configs.
+          schema:
+            $ref: "#/definitions/tenantConfigs"
+        400:
+          description: Indicates invalid parameter was passed.
+          schema:
+            $ref: "#/definitions/errorMessage"
+        404:
+          description: Indicates that there wasn't any custom configs for the provided tenant.
```

Comment:
Better to handle this with an empty array.

Reply:
I'll update the DB store code to return the empty array back to the model layer (pull request [here](https://github.com/CAFDataProcessing/data-processing-service/pull/34)).